packages/web/src/components/pages/Tests/TestDetails/TestSettings/SettingsVariables/Arguments.tsx

```diff
@@ -28,6 +28,14 @@ import {prettifyArguments} from '@utils/prettifyArguments';
 
 import {ArgumentsWrapper} from './Arguments.styled';
 
+const formatArgsArray = (args: string[]) => {
+  return args.map(arg => {
+    const trimmedArg = arg.replace(/\s+/g, ' ').trim();
+
+    return arg.includes(' ') && !/^(['"]).*\1$/.test(arg) ? `"${trimmedArg}"` : trimmedArg;
```

Comment:
Checking the starting character won't work, as:
```yaml
args:
- "some value"
```
Will result in:
```json
["some value"]
```

It will start from `['"]` only in case it specifically has it, and it should be persisted:
```yaml
args:
- "\"abc"
```

Additionally, the quotes inside needs to be escaped too. As an example `abra"kadabra` should not be escaped as `"abra"kadabra"`, but rather `"abra\"kadabra"`